shift_detector/checks/statistical_checks/categorical_statistical_check.py

```diff
@@ -17,14 +20,42 @@ def chi2_test(part1: pd.Series, part2: pd.Series):
 
 class CategoricalStatisticalCheck(SimpleStatisticalCheck):
 
+    def __init__(self, significance=0.01, sample_size=None, use_equal_dataset_sizes=False, sampling_seed=0,
+                 use_binning=False, use_embedding=False):
+        self.use_binning = use_binning
+        self.use_embedding = use_embedding
+        super().__init__(significance, sample_size, use_equal_dataset_sizes, sampling_seed)
+
     def check_name(self) -> str:
         return 'Categorical Statistical Check'
 
     def statistical_test_name(self) -> str:
         return 'Chi^2-Test with Log-Likelihood (G-Test)'
 
     def data_to_process(self, store):
-        df1, df2, columns = store[LowCardinalityPrecalculation()]
+        if not self.use_binning:
+            cat1, cat2, cat_col = store[LowCardinalityPrecalculation()]
+        else:
+            cat1, cat2 = store[ColumnType.categorical]
+            cat_col = store.column_names(ColumnType.categorical)
+        df1 = cat1
+        df2 = cat2
+        columns = set(cat_col)
+        if self.use_binning:
+            num1, num2, num_col = store[BinningPrecalculation(bins=20)]
+            df1 = pd.concat([df1, num1], axis='columns')
+            df2 = pd.concat([df2, num2], axis='columns')
+            columns = columns.union(set(num_col))
+        if self.use_embedding:
+            embedding = LdaEmbedding(store.column_names(ColumnType.text))
+            lda_text = store[embedding]
+            topic_num_to_label = dict(zip(range(embedding.n_topics),
+                                          ['topic_{}'.format(i) for i in range(embedding.n_topics)]))
```

Comment:
This leads to the topics in the plot being wrongly ordered. As topics is already in the plot title, I would leave out the string here

Reply:
I can already here people complain about this...